src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -39,98 +40,234 @@ private ExpressionSyntax ParseInterpolatedStringToken()
-                    // so we still treat it as a composite format string.
-                    var text = originalText[new Range(openQuoteRange.End, closeQuoteRange.Start)];
-                    if (text.Length > 0)
-                    {
-                        builder.Add(SyntaxFactory.InterpolatedStringText(MakeInterpolatedStringTextToken(text, kind)));
-                    }
+                    var interpolation = interpolations[i];
+
+                    // Add a token for text preceding the interpolation
+                    builder.Add(makeContent(
+                        indentationWhitespace, content, first: i == 0, last: false,
+                        originalTextSpan[currentContentStart..interpolation.OpenBraceRange.Start])!);
+
+                    builder.Add(ParseInterpolation(this.Options, originalText, interpolation, kind));
+                    currentContentStart = interpolation.CloseBraceRange.End;
                 }
-                else
+
+                // Add a token for text following the last interpolation
+                builder.Add(makeContent(
+                    indentationWhitespace, content, first: interpolations.Count == 0, last: true,
+                    originalTextSpan[currentContentStart..closeQuoteRange.Start])!);
+
+                CodeAnalysis.Syntax.InternalSyntax.SyntaxList<InterpolatedStringContentSyntax> result = builder;
+                _pool.Free(builder);
+                content.Free();
+                return result;
+            }
+
+            // Gets the indentation whitespace from the last line of a multi-line raw literal.
+            ReadOnlySpan<char> getIndentationWhitespace(ReadOnlySpan<char> originalTextSpan)
+            {
+                // The content we want to create text token out of.  Effectively, what is in the text sections
+                // minus leading whitespace.
+                var closeQuoteText = originalTextSpan[closeQuoteRange];
+
+                // A multi-line raw interpolation without errors always ends with a new-line, some number of spaces, and
+                // the quotes. So it's safe to just pull off the first two characters here to find where the
+                // newline-ends.
+                var afterNewLine = SlidingTextWindow.GetNewLineWidth(closeQuoteText[0], closeQuoteText[1]);
+                var afterWhitespace = SkipWhitespace(closeQuoteText, afterNewLine);
+
+                Debug.Assert(closeQuoteText[afterWhitespace] == '"');
+                return closeQuoteText[afterNewLine..afterWhitespace];
+            }
+
+            InterpolatedStringContentSyntax? makeContent(
+                ReadOnlySpan<char> indentationWhitespace, StringBuilder content, bool first, bool last, ReadOnlySpan<char> text)
+            {
+                if (text.Length == 0)
+                    return null;
+
+                if (!isMultiLineWithoutErrors)
```

Comment:
Consider also exiting quickly when `indentationWhitespace` is empty

Reply:
done!